pkg/controllermanager/controller/backupentry/backup_entry_actuator.go

```diff
@@ -0,0 +1,267 @@
+	"strings"
+	"time"
+
+	"github.com/gardener/gardener/pkg/utils/kubernetes/health"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	gardencorev1alpha1helper "github.com/gardener/gardener/pkg/apis/core/v1alpha1/helper"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kretry "k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultTimeout  = 30 * time.Second
+	defaultInterval = 5 * time.Second
+)
+
+// Actuator acts upon BackupEntry resources.
+type Actuator interface {
+	// Reconcile reconciles the BackupEntry.
+	Reconcile(context.Context) error
+	// Delete deletes the BackupEntry.
+	Delete(context.Context) error
+}
+
+type actuator struct {
+	gardenClient client.Client
+	seedClient   client.Client
+	logger       logrus.FieldLogger
+	backupEntry  *gardencorev1alpha1.BackupEntry
+}
+
+func newActuator(gardenClient, seedClient client.Client, be *gardencorev1alpha1.BackupEntry, logger logrus.FieldLogger) Actuator {
+	return &actuator{
+		logger:       logger.WithField("backupentry", be.Name),
+		backupEntry:  be,
+		gardenClient: gardenClient,
+		seedClient:   seedClient,
+	}
+}
+
+func (a *actuator) Reconcile(ctx context.Context) error {
+	var (
+		g = flow.NewGraph("Backup Entry Creation")
+
+		deployBackupEntryExtension = g.Add(flow.Task{
+			Name: "Deploying backup entry extension resource",
+			Fn:   flow.TaskFn(a.deployBackupEntryExtension).RetryUntilTimeout(defaultInterval, defaultTimeout),
+		})
+
+		_ = g.Add(flow.Task{
+			Name:         "Waiting until backup entry is reconciled",
+			Fn:           flow.TaskFn(a.waitUntilBackupEntryExtensionReconciled),
+			Dependencies: flow.NewTaskIDs(deployBackupEntryExtension),
+		})
+		f = g.Compile()
+	)
+	return f.Run(flow.Opts{
+		Logger:           a.logger,
+		ProgressReporter: a.reportBackupEntryProgress,
+		Context:          ctx,
+	})
+}
+
+func (a *actuator) Delete(ctx context.Context) error {
+	var (
+		g                 = flow.NewGraph("Backup entry deletion")
+		deleteBackupEntry = g.Add(flow.Task{
+			Name: "Destroying backup entry extension",
+			Fn:   flow.TaskFn(a.deleteBackupEntryExtension),
+		})
+		waitUntilBackupEntryExtensionDeleted = g.Add(flow.Task{
+			Name:         "Waiting until extension backup Entry is deleted",
+			Fn:           flow.TaskFn(a.waitUntilBackupEntryExtensionDeleted),
+			Dependencies: flow.NewTaskIDs(deleteBackupEntry),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Delete the secret referred by extension backup Entry",
+			Fn:           flow.TaskFn(a.deleteBackupEntryExtensionSecret),
```

Comment:
Same question as in the `BackupBucket` flow.

Reply:
Will change it.